SSA/Projects/InstCombine/ForLean.lean

```diff
@@ -569,6 +569,26 @@ theorem xor_allOnes_sshiftRight_xor_allOnes {a b : BitVec w} :
     rw [ih]
     rw [BitVec.sshiftRight_add]
 
+@[simp]
+theorem shiftLeft_shiftRight (x : BitVec w) (n : Nat):
+  x >>> n <<< n = x &&& BitVec.allOnes w <<< n := by
+  induction n generalizing x
+  case zero =>
+    simp [bv_toNat]
+  case succ n ih =>
+    rw [BitVec.shiftLeft_add, Nat.add_comm, BitVec.shiftRight_add]
+    rw [ih]
+    rw [Nat.add_comm, BitVec.shiftLeft_add, BitVec.shiftLeft_and_distrib]
+    ext i
+    simp
+    rw [Nat.add_comm]
```

Comment:
Turn this into a `simp only`

Reply:
I tried but it breaks, I'm afraid that's possible since `Nat.add_comm` has no `@[simp]`?